packages/backend-core/src/types/errors.ts

```diff
@@ -0,0 +1,45 @@
+/* Expected values for the 'Auth-Result' response header */
+export enum AuthErrorReason {
+  KeyMissing = 'key-missing',
+  InlineKeyMissing = 'inline-key-missing',
+  InlineKeyInvalid = 'inline-key-invalid',
+  HeaderExpired = 'header-expired',
+  HeaderEarly = 'header-early',
+  HeaderInvalid = 'header-invalid',
+  HeaderInvalidIssuer = 'header-invalid-issuer',
+  HeaderKeyMismatch = 'header-key-mismatch',
```

Comment:
It looks like you introduced "HeaderUnverified" instead of using "HeaderKeyMismatch"

I suggested the name "HeaderKeyMismatch" and "CookieKeyMismatch" because when verification fails, we don't know which mistake the developer made. We've seen both of:
1. They're using a dev instance Frontend API but a prod instance JWT Key
2. They're using a prod instance Frontend API but a dev instance JWT Key

The final possibility is that we're dealing with an attacker.

I thought "mismatch" was a polite way to describe the possible developer error.

I still lean toward "KeyMismatch" but happy to go either way.  We should remove the one we don't choose from the `AuthErrorReason` enum